pkg/operator/ceph/file/controller.go

```diff
@@ -269,3 +299,16 @@ func (c *FilesystemController) releaseOrchestrationLock() {
 	c.orchestrationMutex.Unlock()
 	logger.Debugf("Released lock for filesystem orchestration")
 }
+
+func updateCephFilesystemStatus(cephfilesystem *cephv1.CephFilesystem, status string, context *clusterd.Context) (*cephv1.CephFilesystem, error) {
```

Comment:
Instead of taking the cephfilesystem as an arg, you may need to call `Get()` to get the latest CR, then call `Update()`. Then if it fails you just log the error and you won't return nil to the caller.